index.js

```diff
@@ -4,6 +4,11 @@ module.exports = {
   },
 
   responseObject: function(response) {
+    try {
+      JSON.parse(response.xhr.responseText);
+    } catch(err) {
+      return {invalid_JSON: response.xhr.responseText};
```

Comment:
We can't do this. We need to throw an error if we get invalid JSON. As is, a consumer of this promise would think the contents of the response had been successfully parsed as JSON, even if the response was not JSON.